File: apps/server/api/src/routes/projects.ts
/**
 * Project Management Routes (/api/projects/*).
 * Controller delegating business logic to ProjectService.
 */
import { Hono } from "hono";
import { ProjectService } from "@/api/src/services/project.service.js";

export const projectRoutes = new Hono();
const projectService = new ProjectService();

/**
 * GET /api/projects — List all active and recent server project folders.
 */
projectRoutes.get("/projects", async (c) => {
  const projects = await projectService.listProjects();
  return c.json({
    success: true,
    data: projects,
  });
});

/**
 * POST /api/projects — Add a user-selected folder as a project workspace.
 */
projectRoutes.post("/projects", async (c) => {
  const body = await c.req.json<{ path: string }>();
  if (!body.path) {
    return c.json({ success: false, error: "Field 'path' is required." }, 400);
  }

  try {
    const project = await projectService.addProject(body.path);
    return c.json({
      success: true,
      data: project,
    });
  } catch (err) {
    const errorMsg = err instanceof Error ? err.message : String(err);
    return c.json({ success: false, error: errorMsg }, 400);
  }
});

/**
 * DELETE /api/projects/:id — Remove a project workspace by ID.
 */
projectRoutes.delete("/projects/:id", async (c) => {
  const projectId = c.req.param("id");
  try {
    const deleted = await projectService.deleteProject(projectId);
    if (!deleted) {
      return c.json({ success: false, error: `Project '${projectId}' not found.` }, 404);
    }
    return c.json({
      success: true,
      data: { id: projectId, deleted: true },
    });
  } catch (err) {
    const errorMsg = err instanceof Error ? err.message : String(err);
    return c.json({ success: false, error: errorMsg }, 500);
  }
});

File: apps/server-go/internal/routes/projects.go
// Project management routes (/api/projects/*). Port of
// apps/server/api/src/routes/projects.ts.
package routes

import (
	"fmt"
	"os"
	"path/filepath"
	"strings"

	"github.com/gofiber/fiber/v2"

	"github.com/Adelodunpeter25/console/apps/server-go/internal/services"
)

func registerProjectRoutes(app *fiber.App, projects *services.ProjectService) {
	app.Get("/api/projects", func(c *fiber.Ctx) error {
		list, err := projects.List()
		if err != nil {
			return fail400(c, err)
		}
		return c.JSON(fiber.Map{"success": true, "data": list})
	})

	app.Post("/api/projects", func(c *fiber.Ctx) error {
		var body struct {
			Path string `json:"path"`
		}
		if err := c.BodyParser(&body); err != nil || body.Path == "" {
			return fail400(c, fmt.Errorf("Field 'path' is required."))
		}
		abs, err := filepath.Abs(body.Path)
		if err != nil {
			return fail400(c, err)
		}
		info, err := os.Stat(abs)
		if err != nil || !info.IsDir() {
			return fail400(c, fmt.Errorf("Directory does not exist: %s", abs))
		}
		project, err := projects.Create(services.CreateProjectOptions{
			Name: strings.TrimSpace(filepath.Base(abs)), Dir: abs,
		})
		if err != nil {
			return fail400(c, err)
		}
		return c.JSON(fiber.Map{"success": true, "data": project})
	})

	app.Delete("/api/projects/:id", func(c *fiber.Ctx) error {
		deleted, err := projects.Delete(c.Params("id"))
		if err != nil {
			return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{"success": false, "error": err.Error()})
		}
		if !deleted {
			return c.Status(fiber.StatusNotFound).JSON(fiber.Map{
				"success": false, "error": fmt.Sprintf("Project '%s' not found.", c.Params("id")),
			})
		}
		return c.JSON(fiber.Map{"success": true, "data": fiber.Map{"id": c.Params("id"), "deleted": true}})
	})
}
